### PR TITLE
Fix "Sub-directory Support" in small steps

### LINKED ISSUE #12205: "Sub-directory Support" does not accept top-level (root) URL
Summary: when configured following the sub-directory support instructions, paths like `/librenms/devices` and `/librenms/validate` work fine, but the top level path `/librenms/` gives a 404 error.

To reproduce:

* I created a fresh Ubuntu 20.04 lxd container
* I installed LibreNMS git head (2793585c) exactly as per the [Install-LibreNMS](https://docs.librenms.org/Installation/Install-LibreNMS/) page, using Apache rather than Nginx.
* I went through the database setup pages, then validate, then added `localhost` device.  LibreNMS is working fine at this point.
* I then modified the configuration as per [Sub-directory Support](https://docs.librenms.org/Extensions/Sub-Directory/)

```
--- /etc/apache2/sites-available/librenms.conf.old	2020-10-11 17:40:42.714705144 +0000
+++ /etc/apache2/sites-available/librenms.conf	2020-10-11 17:47:59.622870362 +0000
@@ -1,8 +1,10 @@
+AllowEncodedSlashes On
+Alias /librenms /opt/librenms/html
+
 <VirtualHost *:80>
   DocumentRoot /opt/librenms/html/
   ServerName  librenms.example.com

-  AllowEncodedSlashes NoDecode
   <Directory "/opt/librenms/html/">
     Require all granted
     AllowOverride All
```

```
--- /opt/librenms/html/.htaccess.old	2020-10-11 17:45:44.213581077 +0000
+++ /opt/librenms/html/.htaccess	2020-10-11 17:45:51.393649471 +0000
@@ -20,7 +20,7 @@
 </IfModule>

 RewriteEngine on
-RewriteBase /
+RewriteBase /librenms
 RewriteCond %{REQUEST_FILENAME} !-f
 RewriteCond %{REQUEST_FILENAME} !-d
 RewriteCond %{REQUEST_URI} !\.(js|ico|txt|gif|jpg|png|css|php)
```

```
--- /opt/librenms/.env.old	2020-10-11 17:45:25.301400911 +0000
+++ /opt/librenms/.env	2020-10-11 17:45:58.561717747 +0000
@@ -5,5 +5,5 @@
 DB_USERNAME=librenms
 DB_PASSWORD=password

-#APP_URL=
+APP_URL=/librenms/
 NODE_ID=5f83401f8b0fa
```

```
--- /opt/librenms/config.php.old	2020-10-11 17:44:58.193142622 +0000
+++ /opt/librenms/config.php	2020-10-11 17:46:05.021779276 +0000
@@ -8,7 +8,7 @@

 ### This should *only* be set if you want to *force* a particular hostname/port
 ### It will prevent the web interface being usable form any other hostname
-#$config['base_url']        = "/";
+$config["base_url"] = '/librenms/';

 ### Enable this to use rrdcached. Be sure rrd_dir is within the rrdcached dir
 ### and that your web server has permission to talk to rrdcached.
```

```
systemctl restart apache2
systemctl restart php7.4-fpm
```

Now go to path `/librenms/validate` and it's fine:

image

But go to path `/librenms/` and it gives a 404 error:

image

Note that the LibreNMS logo in the top left-hand corner of the screen links to `/librenms/`

image

### PR DESCRIPTION
**Pull request by mpikzink**
This fixes small parts from #12205 part 3
fixes #12205

Even someone knows a better way, for example to fix js with the url() function. In my case it works.

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/html/pages/oxidized.inc.php [hunk 1]
```diff
@@ -82,7 +82,7 @@
         formatters: {
             "hostname": function(column, row) {
                 if (row.id) {
-                    return '<a href="/device/device=' + row.id + '">' + row.hostname + '</a>';
+                    return '<a href="device/device=' + row.id + '">' + row.hostname + '</a>';
```
Comment (PipoCanaja) [inline includes/html/pages/oxidized.inc.php:85]: Shouldn't the `url()` function be used here ?

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

**Comment by PipoCanaja** [comment]
Hi @mpikzink 
I am currently on holidays and should be able to have a look at this in about 1 week. Looks good "in theory", but I have no way to test it currently. 
Thx

File: includes/html/pages/oxidized.inc.php (hunk 1 above)
Comment (murrant) [reply includes/html/pages/oxidized.inc.php]: The new URL format is `device/<device_id>`  such as `https://librenms.url/device/1`

**Comment by mpikzink** [comment]
Hi @PipoCanaja 
No rush! Whenever you find the time. It's not that urgent. Enjoy your holidays!!!

**Comment by mpikzink** [comment]
@murrant 
Can you please check again if I understood you correctly regarding the URL?

**Comment by murrant** [comment]
Yep, for more info, check the routes file:
https://github.com/librenms/librenms/blob/master/routes/web.php#L109 device
https://github.com/librenms/librenms/blob/master/routes/web.php#L48 tabs
https://github.com/librenms/librenms/blob/master/routes/web.php#L52 device via hostname instead of id

**Review by murrant — APPROVED** [review]
Looks good and tested.

**Pull request merged**